spec/unit/http-api/utils.spec.ts

```diff
@@ -93,6 +93,9 @@ describe("parseErrorResponse", () => {
                     getResponseHeader(name: string): string | null {
                         return name === "Content-Type" ? "application/json" : null;
                     },
+                    getAllResponseHeaders() {
+                        return "Content-Type: application/json";
+                    },
                     status: 500,
                 } as XMLHttpRequest,
                 '{"errcode": "TEST"}',
```

Comment:
would be good to add some more thorough testing of that header-parsing magic here. Eg: what happens if there are multiple header lines? Or if there are `: ` strings in the header values?

Reply:
9ba714df9